src/signalrclient/connection_impl.cpp

```diff
@@ -85,173 +94,177 @@ namespace signalr
             }
 
             // there should not be any active transport at this point
-            _ASSERTE(!m_transport);
+            assert(!m_transport);
 
-            m_disconnect_cts = pplx::cancellation_token_source();
+            m_disconnect_cts = std::make_shared<cancellation_token>();
             m_start_completed_event.reset();
             m_connection_id = "";
         }
 
-        start_negotiate(m_base_url, 0)
-            .then([callback](pplx::task<void> prev_task)
+        start_negotiate(m_base_url, 0, [callback](std::exception_ptr exception)
         {
-            try
+            if (exception != nullptr)
```

Comment:
I like handling the non-exceptional case first like you did in the destructor. I don't like flipping back and forth between handling the exceptional case first or last.